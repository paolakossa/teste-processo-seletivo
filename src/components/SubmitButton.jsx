import React from 'react';
import { Button } from 'antd';
import styled from 'styled-components';

const StyledButton = styled(Button)`
  width: 100%;
`;

export const SubmitButton = () => {
  return (
    <StyledButton type="primary" htmlType="submit">
      Entrar
    </StyledButton>
  );
};
