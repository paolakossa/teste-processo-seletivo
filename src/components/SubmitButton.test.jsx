import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { expect, vi, describe, it } from 'vitest';
import { SubmitButton } from './SubmitButton';

describe('SubmitButton Component', () => {
  it('deve renderizar corretamente', () => {
    render(<SubmitButton />);
    expect(screen.getByRole('button', { name: /entrar/i })).toBeInTheDocument();
  });

  it('deve corresponder ao snapshot', () => {
    const { container } = render(<SubmitButton />);
    expect(container).toMatchSnapshot();
  });

  it('deve enviar o formulário ao clicar no botão', () => {
    const handleSubmit = vi.fn((e) => e.preventDefault());

    render(
      <form onSubmit={handleSubmit}>
        <SubmitButton />
      </form>,
    );

    const button = screen.getByRole('button', { name: /entrar/i });
    fireEvent.click(button);

    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });
});
