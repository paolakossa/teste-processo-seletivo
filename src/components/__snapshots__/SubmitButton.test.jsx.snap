// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubmitButton Component > deve corresponder ao snapshot 1`] = `
.c0 {
  width: 100%;
}

<div>
  <button
    class="ant-btn css-dev-only-do-not-override-tjsggz ant-btn-primary ant-btn-color-primary ant-btn-variant-solid c0"
    type="submit"
  >
    <span>
      Entrar
    </span>
  </button>
</div>
`;
